src/JSURI.purs

```diff
@@ -0,0 +1,90 @@
+module JSURI
+  ( encodeURIComponent
+  , encodeFormURLComponent
+  , decodeURIComponent
+  , decodeFormURLComponent
+  ) where
+
+import Prelude
+
+import Data.Function.Uncurried (Fn3, runFn3)
+import Data.Maybe (Maybe(..))
+import Data.String as String
+
+foreign import _encodeURIComponent :: Fn3 (String -> Maybe String) (String -> Maybe String) String (Maybe String)
+
+-- | URI-encode a string according to RFC3896. Implemented using JavaScript's
+-- | `encodeURIComponent`.
+-- |
+-- | ```purs
+-- | > encodeURIComponent "https://purescript.org"
+-- | Just "https%3A%2F%2Fpurescript.org"
+-- | ```
+-- |
+-- | Encoding a URI can fail with a `URIError` if the string contains malformed
+-- | characters. If you are confident you are encoding a well-formed string then
+-- | you can run this function unsafely:
+-- |
+-- | ```purs
+-- | import Partial.Unsafe (unsafePartial)
+-- | import Data.Maybe (fromJust)
+-- |
+-- | unsafeEncode :: String -> String
+-- | unsafeEncode str = unsafePartial $ fromJust $ encodeURIComponent str
+-- | ```
+encodeURIComponent :: String -> Maybe String
```

Comment:
I also considered `encodeRFC3896` or `encodeURIRFC3896` given that this isn't strictly `encodeURIComponent` from JavaScript, but rather that function augmented to properly conform to the spec.